src/storage/src/hummock/sstable/bloom.rs

```diff
@@ -52,42 +60,105 @@ impl<T: AsMut<[u8]>> BitSliceMut for T {
 }
 
 /// Bloom implements Bloom filter functionalities over a bit-slice of data.
-pub struct Bloom<'a> {
+#[derive(Clone)]
+pub struct BloomFilterReader {
     /// data of filter in bits
-    filter: &'a [u8],
+    data: Vec<u8>,
     /// number of hash functions
     k: u8,
 }
 
-impl<'a> Bloom<'a> {
+impl BloomFilterReader {
     /// Creates a Bloom filter from a byte slice
-    pub fn new(buf: &'a [u8]) -> Self {
-        let filter = &buf[..buf.len() - 1];
+    pub fn new(mut buf: Vec<u8>) -> Self {
+        if buf.len() <= 1 {
+            return Self { data: vec![], k: 0 };
+        }
         let k = buf[buf.len() - 1];
-        Self { filter, k }
+        buf.resize(buf.len() - 1, 0);
+        Self { data: buf, k }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.data.is_empty()
+    }
+
+    /// Judges whether the hash value is in the table with the given false positive rate.
+    ///
+    /// Note:
+    ///   - if the return value is false, then the table surely does not have the user key that has
+    ///     the hash;
+    ///   - if the return value is true, then the table may or may not have the user key that has
+    ///     the hash actually, a.k.a. we don't know the answer.
+    pub fn may_match(&self, mut h: u32) -> bool {
```

Comment:
why rename `surely_not_have_hash` to `may_match` ?

it may conflict with https://github.com/risingwavelabs/risingwave/pull/7489

Reply:
two reason.

1. `may_match` is the function of `RocksDB`.
2. `may_match` is similar to the `hit_sstable_bloom_filter` so it would be easy to understand.

Reply:
Sounds reasonable, if we decide to do this, we need to modify https://github.com/risingwavelabs/risingwave/pull/7489 together